cmd/migration-manager/internal/cmds/target.go

```diff
@@ -164,38 +157,38 @@ func (c *cmdTargetAdd) Run(cmd *cobra.Command, args []string) error {
 			return err
 		}
 
-		internalTarget, err := target.NewInternalIncusTargetFrom(t)
+		// Insert into database.
+		content, err := json.Marshal(t)
 		if err != nil {
 			return err
 		}
 
-		// Verify we can connect to the target, and if using OIDC grab the tokens.
-		if c.additionalRootCertificate != nil {
-			internalTarget.WithAdditionalRootCertificate(c.additionalRootCertificate)
-		}
-
-		if !c.flagNoTestConnection {
-			ctx := context.TODO()
-			err = internalTarget.Connect(ctx)
-			if err != nil {
-				return err
-			}
+		resp, err := c.global.doHTTPRequestV1("/targets", http.MethodPost, "", content)
+		if err != nil {
+			return err
 		}
 
-		// Need to grab properties post-connection to get any OIDC tokens that we need to save.
-		t.Properties = internalTarget.Properties
-
-		// Insert into database.
-		content, err := json.Marshal(t)
+		metadata := make(map[string]string)
+		err = json.Unmarshal(resp.Metadata, &metadata)
 		if err != nil {
 			return err
 		}
 
-		_, err = c.global.doHTTPRequestV1("/targets", http.MethodPost, "", content)
+		connectivityStatusInt, err := strconv.Atoi(metadata["ConnectivityStatus"])
 		if err != nil {
 			return err
 		}
 
+		connectivityStatus := api.ExternalConnectivityStatus(connectivityStatusInt)
+
+		if connectivityStatus == api.EXTERNALCONNECTIVITYSTATUS_TLS_CONFIRM_FINGERPRINT {
+			return fmt.Errorf("Successfully added new target %q, but received an untrusted TLS server certificate with fingerprint %s. Please update the target to correct the issue.", t.Name, metadata["certFingerprint"])
+		} else if connectivityStatus == api.EXTERNALCONNECTIVITYSTATUS_WAITING_OIDC {
+			return fmt.Errorf("Successfully added new target %q; please visit %s to complete OIDC authorization.", t.Name, metadata["OIDCURL"])
```

Comment:
Even if someone does visit the link and completes authorization, they'd still have to run `target update` because the connectivity status won't be updated. 

I think this can be solved by periodically (in `trySyncAllSources` or a new task) to refresh connectivity status if the status is `WAITING_OIDC`.

Reply:
A gofunc is created to handle connectivity status when the user completes the OIDC authentication workflow, so there's no need for an additional `target update` by the user: (https://github.com/FuturFusion/migration-manager/pull/112/files#diff-dce1501c0a00ffd74812cf2f0ba6ea0ca64b3e89819bb109222f8eaec80000c2R247-R261)